modules/programs/qcal.nix

```
@@ -0,0 +1,56 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.programs.qcal;
+
+  qcalAccounts = lib.attrValues
+    (lib.filterAttrs (_: a: a.qcal.enable) config.accounts.calendar.accounts);
+
+  rename = oldname:
+    builtins.getAttr oldname {
+      url = "Url";
+      userName = "Username";
+      passwordCommand = "PasswordCmd";
+    };
+
+  filteredAccounts = lib.trivial.pipe qcalAccounts (map map [
+    # We're only interested in config options in the remote submodule
+    (builtins.getAttr "remote")
+    # Get only the attributes we care about
+    (lib.getAttrs [ "url" "userName" "passwordCommand" ])
+    (lib.filterAttrs (_: attr: !(isNull attr)))
+    # qcal expects these values under other names, so we rename the keys in the attribute set
+    (lib.mapAttrs' (name: value: lib.nameValuePair (rename name) value))
+    # qcal wants the password command as a string, not as a list of strings
+    (lib.mapAttrs (_: value:
+      if lib.isString value then value else lib.concatStringsSep " " value))
+  ]);
```

This seems overly complicated, wouldn't something like
```suggestion
  filteredAccounts = let
    mkAccount = account:
      lib.filterAttrs (_: v: v != null) (with account.remote; {
        Url = url;
        Username = userName;
        PasswordCmd =
          if passwordCommand == null then null else toString passwordCommand;
      });
  in map mkAccount qcalAccounts;
```
suffice?

---

Yes, that is much better - I had come up with what I initially had and didn't quite know how to best improve it. Username could be null as well though. Added that to the PR